presidio-analyzer/presidio_analyzer/entity_recognizer.py

```diff
@@ -76,6 +81,34 @@ def analyze(
         """
         return None
 
+    def enhance_using_context(
+        self,
+        text: str,
+        raw_results: List[RecognizerResult],
+        nlp_artifacts: NlpArtifacts,
+        context: Optional[List[str]] = None,
+    ) -> List[RecognizerResult]:
+        """Enhance confidence score using context of the entity.
+
+        Override this method in derived class in case a custom logic
+        is needed, otherwise return value will be equal to
+        raw_results.
+
+        in case a result score is boosted, derived class need to update
+        result.recognition_metadata[RecognizerResult.IS_SCORE_ENHANCED_BY_CONTEXT_KEY]
+
+        :param text: The actual text that was analyzed
+        :param raw_results: Recognizer results which didn't take
+                            context into consideration
+        :param nlp_artifacts: The nlp artifacts contains elements
+                              such as lemmatized tokens for better
+                              accuracy of the context enhancement process
+        :param context: list of context words
+        """
+        # deep copy and return a new list because its mutable
+        results = copy.deepcopy(raw_results)
```

Comment:
Why is copying needed? I assume it's ok to alter the results based on context. WDYT?

Reply:
removed